tools/biohansel/biohansel.xml

```diff
@@ -40,21 +47,21 @@
 
 ## Create symlinks from Galaxy *.dat to <sample_name>(.fasta|.fastq|.fastq.gz)
 #if $input.type == 'fasta'
-#set $input_files = '"{}"'.format($input.fasta.name)
+  #set $input_files = '"{}"'.format($input.fasta.name)
   ln -s "$input.fasta" $input_files &&
 #elif $input.type == 'paired'
-#set $forward_filename = $get_paired_fastq_filename($input.forward)
-#set $reverse_filename = $get_paired_fastq_filename($input.reverse, is_forward=False)
-#set $input_files = '{} {}'.format($forward_filename, $reverse_filename)
+  #set $forward_filename = $get_paired_fastq_filename($input.forward)
```

Comment:
I'm wondering if, when processing paired-end files, instead of processing the `name` you could make use of the `element_identifier`.  This should be what shows up in the middle section when creating a collection of paired-end files (or a single pair of files):

![image](https://user-images.githubusercontent.com/200517/43603222-f6dc5176-9657-11e8-97bb-5a9aa2113dcd.png)

The reason for using this identifier is that, in IRIDA, we set this value to be the sample name. This could be different from the sample name derived from the fastq file name (in the case where a user re-names a sample).

I think with the code you have now you could do this by setting `$name` to `$input.element_identifier`.

```
#set $forward_filename = $get_paired_fastq_filename($input.forward, $input.element_identifier)
#set $reverse_filename = $get_paired_fastq_filename($input.reverse, $input.element_identifier, is_forward=False)
```

Could you also do this for the **paired_collection** section?

Reply:
Hi Aaron,

It looks like using the `name` attribute in 2.0.0 of the Galaxy tool wrapper was working with IRIDA.

I ran biohansel through IRIDA where I named the sample *sample* and the files `SRR5646583_S1_L001_R1_001.fastq` and `SRR5646583_S1_L001_R2_001.fastq`.

The command in Galaxy was:

```bash
ln -s ".../irida-data/sequence/14/1/SRR5646583_S1_L001_R1_001.fastq" "sample_1.fastq" && \
ln -s ".../irida-data/sequence/13/1/SRR5646583_S1_L001_R2_001.fastq" "sample_2.fastq" && \
hansel -vvv -t "${GALAXY_SLOTS:-1}" -o results.tab -O match_results.tab -S tech_results.tab --json "sample_1.fastq" "sample_2.fastq" --scheme heidelberg --min-kmer-freq 8 --max-kmer-freq 1000 --low-cov-depth-freq 20 --max-missing-tiles 0.05 --min-ambiguous-tiles 3 --max-intermediate-tiles 0.05 --low-cov-warning 20
```

And the output had the correct sample name *sample*:

```
sample	scheme	scheme_version	subtype	all_subtypes	tiles_matching_subtype	are_subtypes_consistent	inconsistent_subtypes	n_tiles_matching_all	n_tiles_matching_all_expected	n_tiles_matching_positive	n_tiles_matching_positive_expected	n_tiles_matching_subtype	n_tiles_matching_subtype_expected	file_path	avg_tile_coverage	qc_status	qc_message
sample	heidelberg	0.5.0	2.2.1.1.1.1	2; 2.2; 2.2.1; 2.2.1.1; 2.2.1.1.1; 2.2.1.1.1.1	1983064-2.2.1.1.1.1; 4211912-2.2.1.1.1.1	True		202	202	20	20	2	2	['sample_1.fastq', 'sample_2.fastq']	42.631	PASS	
```


`$input.paired_collection` only had a `name` attribute and no `element_identifier` attribute. When creating the paired collection in Galaxy and with the IRIDA created paired collection, `$input.paired_collection.forward.element_identifier` was `forward` and `$input.paired_collection.reverse.element_identifier` was `reverse`. 

When running biohansel directly in Galaxy while specifying the forward and reverse reads files individually, the `element_identifier` and the `name` were the same with `$input.forward` and `$input.reverse`. When exporting to Galaxy does IRIDA set `element_identifier` to be the Sample name for each reads file?

Reply:
Thanks for checking this out Peter.

So, in IRIDA it looks like I set the `name` attribute [here](https://github.com/phac-nml/irida/blob/75c784adae046ae765cc0eb8f32b8c163923193a/src/main/java/ca/corefacility/bioinformatics/irida/service/analysis/workspace/galaxy/AnalysisCollectionServiceGalaxy.java#L160).  I don't think there's any other attribute I can set (so I can't set `element_identifier`).

If everything works via IRIDA (as you've verified) and it's working okay in Galaxy then I would say what you have now is good and there's no need to change anything.  So the above comment can be ignored.